app/data_model/answer_store.py

```diff
@@ -0,0 +1,135 @@
+import re
+
+from collections import OrderedDict
+
+
+class AnswerStore(object):
+    """
+    An object that stores the flattened structure of AnswerStates.
+    It is referenced by the answers property in the QuestionnaireStore.
+    """
+
+    def __init__(self, existing_answers=None):
+        self.answers = existing_answers or []
+
+    def add(self, answer):
+        """
+        Add a new answer into the answer store.
+
+        :param answer: A dict of flattened answer details.
+        :return: None.
+        """
+        answer_to_add = answer.copy()
+
+        if self.exists(answer_to_add):
+            self.update(answer)
+        else:
+            self.answers.append(answer_to_add)
+
+    def update(self, answer):
+        """
+        Update the value of an answer already in the answer store.
+
+        :param answer: A dict of flattened answer details.
+        :return:
+        """
+        for index, existing in enumerate(self.answers):
+            if self.same(existing, answer):
+                self.answers[index]['value'] = answer['value']
+                break
+
+    def get(self, answer):
+        """
+        Returns the value of an answer.
+
+        :param answer:The alue of an answer
+        :return:
+        """
+        found = False
+
+        for existing in self.answers:
+            if self.same(answer, existing):
+                found = True
```

Comment:
self.exists also loops the entire set of answers, so self.same should be more performant.